ai.d.lua

```diff
@@ -8,25 +8,46 @@
 ---@module 'main'
 ---@module 'math'
 
----@alias AiMovementMode
----| '"Off"'
----| '"Manual"'
----| '"Automatic"'
----| '"Fleet"'
-
----@alias AiFiringMode
----| '"Off"'
----| '"On"'
+---@enum AiMovementModes
+local AiMovementMode = {
+	Off = 0,
+	Manual = 1,
+	Automatic = 2,
+	Fleet = 3
+}
 
 ---@class AI
 local ai = {}
 
 ---@param index integer index of the AI mainframe
----@return AiMovementMode movement mode of the AI mainframe
-function ai.getMovementMode(index)
+---@return AiMainframe? the AI at the given index, or nil if it doesn't exist
+function ai.get(index)
+
 end
 
----@param index integer index of the AI mainframe
----@return AiMovementMode firing mode of the AI mainframe
-function ai.getFiringMode(index)
-end
\ No newline at end of file
+-- This corresponds directly with the breadboard "AI Behaviour Input"
+--
+-- Note: each of these fields will need a __set to make C# reflect these changes,
```

Comment:
these fields will be handled by get_name and set_name functions, which the object base "class" calls for assignment and access automatically